src/Models/NonhydrostaticModels/pressure_correction.jl

```diff
@@ -33,10 +33,10 @@ Update the predictor velocities u, v, and w with the non-hydrostatic pressure vi
 """
 @kernel function _pressure_correct_velocities!(U, grid, Δt, pNHS)
     i, j, k = @index(Global, NTuple)
-
-    @inbounds U.u[i, j, k] -= ∂xᶠᶜᶜ(i, j, k, grid, pNHS) * Δt
-    @inbounds U.v[i, j, k] -= ∂yᶜᶠᶜ(i, j, k, grid, pNHS) * Δt
-    @inbounds U.w[i, j, k] -= ∂zᶜᶜᶠ(i, j, k, grid, pNHS) * Δt
+    
+    @inbounds U.u[i, j, k] -= δxᶠᵃᵃ(i, j, k, grid, pNHS) / Δxᶠᶜᶜ(i, j, k, grid) * Δt
+    @inbounds U.v[i, j, k] -= δyᵃᶠᵃ(i, j, k, grid, pNHS) / Δyᶜᶠᶜ(i, j, k, grid) * Δt
+    @inbounds U.w[i, j, k] -= δzᵃᵃᶠ(i, j, k, grid, pNHS) / Δzᶜᶜᶠ(i, j, k, grid) * Δt
```

Comment:
So the idea is that while `∂x` operators dispatch on IBGrids (and are thus zero across immersed boundaries), `δx` operators do not, right?

If so, I can then see how this change leads to a divergence-free velocity field, but I don't see how impenetrability can be satisfied everywhere...

Do you understand this behavior?